src/test/java/it/sweven/blockcovid/routers/AdminRouterTest.java

```diff
@@ -78,22 +79,40 @@ void setUp() {
   }
 
   @Test
-  void register() {
-    Credentials testCredentials = new Credentials("test", "test", Set.of(Authority.USER));
-    String auth = "testAuthorization";
-    User testUser =
-        new UserBuilder()
-            .setUsername(testCredentials.getUsername())
-            .setPassword(testCredentials.getPassword())
-            .setAuthorities(Set.of(Authority.USER))
-            .createUser();
-    User testAdmin =
-        new UserBuilder()
-            .setUsername("testUser")
-            .setPassword("testPassword")
-            .setAuthorities(Set.of(Authority.ADMIN))
-            .createUser();
-    // assertEquals(userAssembler.toModel(testUser), router.register(testCredentials, auth));
+  void register_validRequest() throws CredentialException {
+    Credentials testCredentials = new Credentials("user", "password", Set.of(Authority.USER));
+    User testUser = new User("user", "password", Set.of(Authority.USER));
+    User testAdmin = new User("admin", "password", Set.of(Authority.ADMIN));
+    when(authenticationService.authenticateByToken("auth")).thenReturn(testAdmin);
+    when(registrationService.register(any())).thenReturn(testUser);
+    assertEquals(userAssembler.toModel(testUser), router.register(testCredentials, "auth"));
+  }
+
+  @Test
+  void register_wrongToken_throwsAuthenticationCredentialsException() {
+    Credentials testCredentials = new Credentials("user", "password", Set.of(Authority.USER));
+    when(authenticationService.authenticateByToken("auth"))
+        .thenThrow(new AuthenticationCredentialsNotFoundException(""));
+    assertThrows(
+        AuthenticationCredentialsNotFoundException.class,
+        () -> router.register(testCredentials, "auth"));
+  }
+
+  @Test
+  void register_usernameAlreadyInUse_throwsResponseStatusException() throws CredentialException {
+    Credentials testCredentials = new Credentials("user", "password", Set.of(Authority.USER));
+    User adminTest = new User("admin", "password", Set.of(Authority.ADMIN));
+    when(authenticationService.authenticateByToken("auth")).thenReturn(adminTest);
+    when(registrationService.register(any())).thenThrow(new CredentialException());
+    assertThrows(ResponseStatusException.class, () -> router.register(testCredentials, "auth"));
```

Comment:
Status returned by `ResponseStatusException` should also be tested.

Same goes for test `register_nullCredentials_throwsResponseStatusException()`